parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -526,7 +528,35 @@ public PageReadStore readNextRowGroup() throws IOException {
     return columnChunkPageReadStore;
   }
 
+  public static DictionaryPage readDictionary(Configuration conf, ColumnChunkMetaData meta, FSDataInputStream fin) throws IOException {
+    if(fin.getPos() != meta.getStartingPos()) {
+      fin.seek(meta.getStartingPos());
+    }
+
+    PageHeader pageHeader = Util.readPageHeader(fin);
+
+    return readDictionary(conf, pageHeader, meta, fin);
+  }
+
+  public static DictionaryPage readDictionary(Configuration conf, PageHeader pageHeader, ColumnChunkMetaData meta, FSDataInputStream fin) throws IOException {
+    DictionaryPageHeader dictHeader = pageHeader.getDictionary_page_header();
+
+    int uncompressedPageSize = pageHeader.getUncompressed_page_size();
+    int compressedPageSize = pageHeader.getCompressed_page_size();
+
+    byte [] dictPageBytes = new byte[compressedPageSize];
+
+    fin.readFully(dictPageBytes);
 
+    BytesInput bin = BytesInput.from(dictPageBytes);
+
+    DictionaryPage compressedPage = new DictionaryPage(bin, uncompressedPageSize, dictHeader.getNum_values(), converter.getEncoding(dictHeader.getEncoding()));
+
+    BytesDecompressor decompressor = new CodecFactory(conf).getDecompressor(meta.getCodec());
+    DictionaryPageReader dictionaryPageReader = new ColumnChunkDictionaryPageReader(decompressor, compressedPage);
```

Comment:
I don't see much value in having a ColumnChunkDictionaryPageReader. That class and associated interfaces aren't needed if you add this to decompress inline:
```java

    try {
      return new DictionaryPage(
          decompressor.decompress(compressedPage.getBytes(), compressedPage.getUncompressedSize()),
          compressedPage.getDictionarySize(),
          compressedPage.getEncoding());
    } catch (IOException e) {
      throw new RuntimeException(e);
    }
```

I do think there is some value to having a RowGroupDictionaryPageReader (as I suggested already) so I think we should just rework those classes a bit.